src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/DuplicateTypeNameAnalyzer.cs

```diff
@@ -148,11 +148,23 @@ public override void Analyze(ISymbolAnalysisContext context)
                 var currentFullTypeName = namedTypeSymbol.ToDisplayString();
                 var currentAssemblyName = namedTypeSymbol.ContainingAssembly?.Name ?? "";
                 
-                // Don't report if it's exactly the same type
-                if (string.Equals(currentFullTypeName, qualifiedTypeName, StringComparison.Ordinal) && 
-                    string.Equals(currentAssemblyName, packageName, StringComparison.Ordinal))
+                // For generic types, we need to check the base type name without generic parameters
+                var currentTypeNameWithoutGenerics = currentFullTypeName;
+                var genericMarkerIndex = currentFullTypeName.IndexOf('<');
+                if (genericMarkerIndex >= 0)
                 {
-                    return;
+                    currentTypeNameWithoutGenerics = currentFullTypeName.Substring(0, genericMarkerIndex);
```

Comment:
why are the comparisons done without generics? I dont want the analyzer to flag `Azure.Operation` and `Azure.Operation<T>` as duplicates.